### DISCUSSION TITLE
Help to set up cask contributing

### DISCUSSION THREAD

**Discussion by eloyaldea** [Homebrew/discussions#500] [Casks]
Hi! I'm trying to set up cask contributing using `brew bump-cask-pr`. I once used `cask-repair` but seeing it's now being abandoned I want to setup the newer way. I have read the guide at [https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md](url) but I can't make it work.
My Git / GitHub / coding knowledge is fairly limited and I know there are some basic things that I don't understand. Could anyone explain it with basic steps from the beginning?
I'd would love to contribute to the community but I'm unable to make it work. The help is appreciated!!

**Comment by carlocab** [comment]
Not sure if I'm forgetting a step here, but, if I'm not:

1. Create a GitHub API token using the instructions [here](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token). Make sure it has appropriate edit permissions.
2. In your shell, do

        export HOMEBREW_GITHUB_API_TOKEN=<token created in step 1>

4. Do `bump-cask-pr`.

You may wish to make sure your `HOMEBREW_GITHUB_API_TOKEN` is set in your `.zshrc` somehow. However, if you have your dotfiles under version control, you'll want to keep your actual token out of version control.

**Reply by eloyaldea** [reply]
Does it need to have a special name? Which permissions does it need?

**Reply by gromgit** [reply]
> Does it need to have a special name?

You're exporting the token _value_, not the name, so "Supercalifragilisticexpialidocious" works as well as "homebrew".

> Which permissions does it need?

Last I checked, `public_repo` and `gist` are the bare minimum.

**Reply by carlocab** [reply]
>"Supercalifragilisticexpialidocious" works as well as "homebrew"

This is false, "supercalifragilisticexpialidocious" works much better

**Reply by eloyaldea** [reply]
Great it works! Thanks a lot but I have another question, the cask 'macupdater' is ironically outdated in Homebrew. The problem is that the version is still the same: 2.0.0, but the build number and SHA256 have changed, and it's not letting me create a pull request with `brew bump-cask-pr` . How can I make it? I tried using the flag `--force` and nope.
I tried to use the flag `--sha256`instead of `--version` and I assumed the SHA256 is obtained on Terminal like this: `shasum -a 256` plus the path to the .dmg downloaded from [http://corecode.io/macupdater/index.html](url). Is there anyway to do this?

**Reply by gromgit** [reply]
First check with upstream _why_ it was changed without a version bump. Cask checksums are always checked before checkins, and [their RSS feed](https://www.corecode.io/macupdater/macupdater2.xml):
```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:content="http://purl.org/rss/1.0/modules/content/" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:sparkle="http://www.andymatuschak.org/xml-namespaces/sparkle" xmlns:atom="http://www.w3.org/2005/Atom">
	<channel>
		<atom:link href="https://www.corecode.io/macupdater/macupdater.xml" rel="self" type="application/rss+xml" />
		<title>MacUpdater Appcast</title>
		<link>https://www.corecode.io/macupdater/index.html</link>
		<description>Contains information about MacUpdater releases.</description>
		<language>en</language>


		<item>
			<title>MacUpdater 2.0.0</title>
			<guid isPermaLink="false">2.0.0</guid>
			<pubDate>Sat, 09 Jan 2021 11:36:46 +0000</pubDate>
			<sparkle:releaseNotesLink>https://www.corecode.io/macupdater/history2.html</sparkle:releaseNotesLink>
			<sparkle:minimumSystemVersion>10.14.0</sparkle:minimumSystemVersion>
			<enclosure	url="https://www.corecode.io/downloads/macupdater_2.0.0.dmg"
						length="12003791"
						sparkle:version="10776"
						sparkle:shortVersionString="2.0.0"
						sparkle:dsaSignature="MCwCFAuFlsBeTLdCpCj2L/ca90OYlodsAhQ7l615BtFsX2uJ/zF97zfDbqm4ug=="
						sparkle:edSignature="+D/Nuc3oc2rqyCMA7K5sj5i5MGLSDB0iSALLW7zSENc3Plhbe3H588rDfy0/zHZU3YZG7yMzI26EkvrBEwN8DA=="
						type="application/octet-stream">
			</enclosure>
		</item>
		
<!-- 
		<item>
			<title>MacUpdater 2.0.0</title>
			<guid isPermaLink="false">2.0.0</guid>
			<pubDate>Tue, 05 Jan 2021 01:19:04 +0000</pubDate>
			<sparkle:releaseNotesLink>https://www.corecode.io/macupdater/history2.html</sparkle:releaseNotesLink>
			<sparkle:minimumSystemVersion>10.14.0</sparkle:minimumSystemVersion>
			<enclosure	url="https://www.corecode.io/downloads/macupdater_2.0.0.dmg"
						length="11755996"
						sparkle:version="10776"
						sparkle:shortVersionString="2.0.0"
						sparkle:dsaSignature="MCwCFHrED/OlgUnd5ZopQwGFsAup+rRdAhQuUYBGUSuZ11H4zl2ybQdx3OQ8oA=="
						sparkle:edSignature="Ky77rxQh6mgXDU9RaUbOmY/CeWfYct+ub81/tthhkhTmZz4hKwlL5F0MhAiXGmQgnva1sXYC4C5f538iE7KbAQ=="
						type="application/octet-stream">
			</enclosure>
		</item>
 -->
	
	</channel>
</rss>
```
clearly indicates that someone changed the DMG on Jan 9, after Homebrew's 2.0.0 cask bump on Jan 6.

**Reply by eloyaldea** [reply]
Um I'm not sure what this is tbh 😅 I don't understand this much about Git sorry. I was trying to use `brew bump-cask-pr` so the version is updated and correct on Homebrew. Because right now it doesn't let you install the cask so I want to fix it. This is what it shows when you run `brew install macupdater`:

> ==> Downloading https://www.corecode.io/downloads/macupdater_2.0.0.dmg
Already downloaded: /Users/XXX/Library/Caches/Homebrew/downloads/9e9d5cf081ec40616b08360abc8ecf547684d76c6b35660ee98584704142d44d--macupdater_2.0.0.dmg
Error: SHA256 mismatch
Expected: 8842c80d3945ac552f2ce2285dc2c98fd872655240b69aa8ae6f8e032d5f487c
  Actual: 0504435473556c47bb299af253fdd74f6b0e1280a09c54843c0729eaf790d79f
    File: /Users/XXX/Library/Caches/Homebrew/downloads/9e9d5cf081ec40616b08360abc8ecf547684d76c6b35660ee98584704142d44d--macupdater_2.0.0.dmg
To retry an incomplete download, remove the file above.

The old .dmg has the version 2.0.0 and build 10776 and the new one has version 2.0.0 and build 10834. The cask [https://github.com/Homebrew/homebrew-cask/blob/master/Casks/macupdater.rb](url) says the version is 2.0.0,10776 so I tried to `brew bump-cask-pr --version 2.0.0,10834 macupdater` but it doesn't work :/.

What can I do next? Thanks!

I replaced my user with XXX.

**Reply by gromgit** [reply]
> I tried to `brew bump-cask-pr --version 2.0.0,10834 macupdater` but it doesn't work

You'll have to be a lot more specific than "it doesn't work". What's the exact output?

**Reply by eloyaldea** [reply]
Sorry, here is the output:

> brew bump-cask-pr --version 2.0.0,10834 macupdater
Error: GitHub Bad credentials:HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
  https://github.com/settings/tokens

I'm confused because my token works as I've made pull request for cask updates as soon as I got it working today.

**Reply by gromgit** [reply]
> I'm confused because my token works as I've made pull request for cask updates as soon as I got it working today.

Are you sure about that? [Your Cask PR record](https://github.com/Homebrew/homebrew-cask/pulls?q=is%3Apr+author%3Aeloyaldea) stops in Jul 2020.

Regardless:
* Have you checked https://github.com/settings/tokens to see if it's expired or deleted or something?
* Have you `echo $HOMEBREW_GITHUB_API_TOKEN` to doublecheck that it's actually a token?

**Reply by eloyaldea** [reply]
I created this PR this morning: [https://github.com/Homebrew/homebrew-cask-versions/pull/10195](url) and it worked fine. On [https://github.com/settings/tokens](url) my token shows 'Last used within the last week'.
I have done `echo $HOMEBREW_GITHUB_API_TOKEN` and it shows a long string of characters how do I verify it's a token?

**Reply by gromgit** [reply]
The only way I know of to verify a token is to use it, and since you seem to have done so successfully, I'm out of ideas.

**Reply by eloyaldea** [reply]
Okay, well anyway thanks for the help! I'll contribute with new cask versions when I can :)
